renderdoc/driver/gl/gl_renderstate.cpp

```diff
@@ -633,7 +633,7 @@ void GLRenderState::MarkDirty(WrappedOpenGL *driver) const
       if(type == eGL_RENDERBUFFER)
         manager->MarkDirtyResource(RenderbufferRes(ctx, name));
       else
-        manager->MarkDirtyResource(TextureRes(ctx, name));
+        manager->MarkDirtyWithWriteReference(TextureRes(ctx, name));
```

Comment:
This function `GLRenderState::MarkDirty` is unfortunately called _after_ the drawcall in all of the places in `gl_draw_funcs.cpp` so that ordering will need to change to allow the last second fetch of postponed resource contents to happen at the right time.

Reply:
Yeah, since we realized this after initial implementation, we should move this one too. I'll work on it today.